tchannel/schemes/thrift.py

```diff
@@ -18,9 +18,11 @@ def __init__(self, tchannel):
         self._tchannel = tchannel
 
     @gen.coroutine
-    def __call__(self, request=None, headers=None, timeout=None,
+    def __call__(self, request, headers=None, timeout=None,
                  retry_on=None, retry_limit=None):
 
+        assert request, "a ThriftRequest is required"
```

Comment:
Actually for this case an ABC makes sense since the isinstance will be cheap and we'll be able to know it's a thrift response.